Renderer/CKEditorRenderer.php

```diff
@@ -25,16 +24,48 @@
 class CKEditorRenderer implements CKEditorRendererInterface
 {
     /**
-     * @var ContainerInterface
+     * @var JsonBuilder
      */
-    private $container;
+    private $jsonBuilder;
 
     /**
-     * @param ContainerInterface $container
+     * @var RouterInterface
      */
-    public function __construct(ContainerInterface $container)
-    {
-        $this->container = $container;
+    private $router;
+
+    /**
+     * @var Packages
+     */
+    private $packages;
+
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    /**
+     * @var string|null
+     */
+    private $locale;
+
+    public function __construct(
+        JsonBuilder $jsonBuilder,
+        RouterInterface $router,
+        RequestStack $requestStack,
+        Packages $packages = null,
+        EngineInterface $templating = null,
+        $locale = null
+    ) {
+        $this->jsonBuilder = $jsonBuilder;
+        $this->router = $router;
+        $this->packages = $packages;
+        $this->templating = $templating;
+        $request = $requestStack->getMasterRequest();
+        if ($request && null !== $request->getLocale()) {
+            $this->locale = $request->getLocale();
+        } else {
```

Comment:
good job on stripping the container and doing a proper dependency construction - if i may add, we could strip this else and make it non-else free if you just set the locale to the $locale before the if.

Reply:
Well I picked up from #340 because I had no idea when it would be finished, but so far we ended up in more or less the same place :P As for removing `if/else`, it was how I initially did it, but decided against setting a value I would overwrite in the next line. This way you have a clear execution path (pedantic, I know).